rsc-server/src/org/openrsc/model/Mob.java

```diff
@@ -0,0 +1,144 @@
+package org.openrsc.model;
+
+/**
+ * A dynamic mobile entity with combat attributes.
+ */
+public abstract class Mob {
+
+	/**
+	 * A unique id generated for the mob's current session.
+	 */
+	private final int sessionId;
+	
+	/**
+	 * The mob's current location.
+	 */
+	private Location location;
+
+	/**
+	 * The mob's last known location.
+	 */
+	private Location travelBack = new Location(-1, -1);
+	
+	// FIXME Figure out is Dan has a plan for these variables.
+    public int admin;
+    public long nameLong;
+    public String name;
+    public int serverIndex;
+    public int mobIntUnknown;
+    public int x;
+    public int z;
+    public int type;
+    public int stepCount;
+    public int currentSprite;
+    public int nextAnimation;
+    public int movingStep;
+    public int waypointCurrent;
+    public int waypointsX[] = new int[10];
+    public int waypointsZ[] = new int[10];
+    public int animationCount[] = new int[12];
+    public String lastMessage;
+    public int lastMessageTimeout;
+    public int anInt162;
+    public int anInt163;
+    public int damageTaken;
+    public int healthCurrent;
+    public int healthMax;
+    public int combatTimer;
+    public int level = -1;
+    public int colourHairType;
+    public int colourTopType;
+    public int colourBottomType;
+    public int colourSkinType;
```

Comment:
Good to know!

Reply:
Can you confirm? Because at first glance, I would assume this was only for players, since players have the ability to change their appearance, and npcs are static assets.

Reply:
Sorry, when I said NPCs I was thinking of humanoid NPCs. Surely they are not static assets? But yes, this will be needed for players.